tests/unit/disco/containers/TestDiscoPane.js

```diff
@@ -1,158 +1,172 @@
     });
   });
 
-  describe('loadDataIfNeeded', () => {
-    it('does nothing if there are loaded results', () => {
-      const store = {
-        getState() {
-          return { addons: { foo: {} }, discoResults: [{ addon: 'foo' }] };
-        },
-      };
-      const getAddons = sinon.stub(discoApi, 'getDiscoveryAddons');
-      return helpers.loadDataIfNeeded({ store })
-        .then(() => expect(getAddons.called).toBeFalsy());
+  describe('mapStateToProps', () => {
+    it('sets extension results', () => {
+      const addon = { ...fakeDiscoAddon };
+
+      const props = helpers.mapStateToProps(loadDiscoResultsIntoState([{
+        heading: 'The Add-on',
+        description: 'editorial text',
+        addon,
+      }]));
+
+      expect(props.results).toEqual([{
+        ...addon,
+        addon: addon.guid,
+        description: 'editorial text',
+        heading: 'The Add-on',
+        iconUrl: addon.icon_url,
+      }]);
     });
 
-    it('loads the addons if there are none', () => {
-      const api = { the: 'config' };
-      const dispatch = sinon.spy();
-      const store = {
-        dispatch,
-        getState() {
-          return { addons: {}, api, discoResults: [] };
-        },
+    it('sets theme results', () => {
+      const addon = {
+        ...fakeDiscoAddon,
+        theme_data: {},
+        type: ADDON_TYPE_THEME,
       };
-      const entities = {
-        addons: {
-          foo: {
-            slug: 'foo',
-          },
-        },
-        discoResults: {
-          foo: {
-            addon: 'foo',
-          },
-        },
-      };
-      const result = { results: ['foo'] };
-      const getAddons = sinon.stub(discoApi, 'getDiscoveryAddons')
-        .returns(Promise.resolve({ entities, result }));
-      return helpers.loadDataIfNeeded({ store })
-        .then(() => {
-          expect(getAddons.calledWith({ api })).toBeTruthy();
-          expect(dispatch.calledWith(loadEntities(entities))).toBeTruthy();
-          expect(dispatch.calledWith(discoResults([{ addon: 'foo' }]))).toBeTruthy();
-        });
-    });
-  });
 
-  describe('mapStateToProps', () => {
-    it('only sets results', () => {
-      const props = helpers.mapStateToProps({
-        discoResults: [],
-      });
-      expect(Object.keys(props)).toEqual(['results']);
-    });
+      const props = helpers.mapStateToProps(loadDiscoResultsIntoState([{
+        heading: 'The Theme',
+        description: 'editorial text',
+        addon,
+      }]));
 
-    it('sets the results', () => {
-      const props = helpers.mapStateToProps({
-        addons: { one: { slug: 'one' }, two: { slug: 'two' } },
-        discoResults: [{ addon: 'two' }],
-        infoDialog: {},
-      });
-      expect(props.results).toEqual([{ slug: 'two', addon: 'two' }]);
+      // This is removed by the reducer.
+      delete addon.theme_data;
+
+      // This is a magical theme guid created by the reducer for some reason.
```

Comment:
Firefox appends the @personas.mozilla.org bit internally. That's how you look it up with mozAddonManager.